packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -300,8 +304,36 @@ public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
 
   private void handleChoosePictureResult(int resultCode, Intent data) {
     if (resultCode == Activity.RESULT_OK && data != null) {
-      String path = fileUtils.getPathFromUri(activity, data.getData());
-      handleResult(path);
+        //String path = fileUtils.getPathFromUri(activity, data.getData());
+
+        //creating a temp file from uri to allow reading pictures from remote sources
+        try {
+            InputStream inputStream = null;
+            OutputStream output = null;
+            try {
```

Comment:
Please use try-with-resources. It is [too hard](https://romain.vernoux.fr/a-try-with-resources-equivalent-for-java-6-and-android/) to get this code correct without.

Reply:
isn't resource management only for API 19+ and flutter for 16+ ?

Reply:
You are right, sorry. In that case, you need to wrap your calls to `close` in try-catch blocks. Otherwise we risk not actually closing the input stream, as well as exiting abnormally from a finally block.